src/cfdcore_schnorrsig.cpp

```diff
@@ -312,6 +314,51 @@ Pubkey SchnorrUtil::ComputeSigPoint(
   return ConvertSecpPubkey(secp_sigpoint);
 }
 
+Pubkey SchnorrUtil::ComputeSigPointBatch(
+    const std::vector<ByteData256> &msgs,
+    const std::vector<SchnorrPubkey> &nonces, const SchnorrPubkey &pubkey) {
+  if (msgs.size() != nonces.size() || msgs.empty()) {
+    throw CfdException(
+        CfdError::kCfdIllegalArgumentError,
+        "Expected same number of messages or nonces, and at least one "
+        "message.");
+  }
+
+  std::vector<secp256k1_xonly_pubkey> secp_nonces;
+  Pubkey rs;
+
+  if (msgs.size() == 1) {
+    rs = Pubkey(ByteData("02").Concat(nonces[0].GetData()));
+  } else {
+    std::vector<Pubkey> pub_nonces;
+    for (size_t i = 0; i < msgs.size(); i++) {
+      pub_nonces.push_back(Pubkey(ByteData("02").Concat(nonces[i].GetData())));
```

Comment:
It seemed to be good for less than that:
```suggestion
    for (const auto& nonce : nonces) {
      pub_nonces.push_back(Pubkey(ByteData("02").Concat(nonce.GetData())));
```

Reply:
Indeed :)